---

- block:
    - name: "{{ TestURL }}- URL to check is as follows:"
      set_fact:
        url_2b_verify: "{{ url2bTested }}"

    - name: "{{ TestURL }}- Verifying the url {{ url_2b_verify }}"
      win_uri:
        url: "{{ url_2b_verify }}"
        validate_certs: no
      register: validate_result
      retries: 2
      delay: 2
      failed_when: false
      until: not validate_result.msg | regex_search('Unable to connect to the remote server')

    - debug:
        var: validate_result

    - name: "{{ TestURL }}- Report the status of BillPayment Application"
      fail:
        msg: "{{ TestURL }}- BillPayment Application is down!"
      when: validate_result.msg | regex_search('Unable to connect to the remote server')

    - debug:
        msg: "{{ TestURL }}- BillPayment Application is up!"
      when: not validate_result.msg | regex_search('Unable to connect to the remote server')

  tags:
  - A13
  - B11

